src/olympia/reviewers/models.py

```diff
@@ -1338,17 +1338,24 @@ def check_is_locked(cls, version):
 
     @classmethod
     def check_has_auto_approval_disabled(cls, version):
-        """Check whether the add-on has auto approval fully disabled by a
-        reviewer (only applies to listed) or disabled temporarily because they
-        had their previous version on a delayed rejection (only applies to
-        listed) or automated scanners (applies to every channel).
+        """Check whether the add-on has auto approval disabled or delayed.
+
+        It could be:
+        - Disabled by a reviewer (different flag for listed or unlisted)
+        - Disabled until next manual approval (only applies to listed, typically
+          set when a previous version is on a delayed rejection)
+        - Delayed until a future date by scanners (applies to both listed and
+          unlisted)
         """
         addon = version.addon
         is_listed = version.channel == amo.RELEASE_CHANNEL_LISTED
-        auto_approval_disabled = is_listed and bool(
-            addon.auto_approval_disabled
-            or addon.auto_approval_disabled_until_next_approval
-        )
+        if is_listed:
```

Comment:
Can an add-on move from listed to unlisted and vice versa? If so, and an add-on was unlisted, and switches to be listed, would the `auto_approval_disabled_unlisted` no longer be valid? The reason I am asking these questions is because I am wondering if, for this logic, you even need to check `is_listed` at all, or whether it could simply be an `or` between `auto_approval_disabled_unlisted`, `auto_approval_disabled` and `auto_approval_disabled_until_next_approval`?

Reply:
I guess the test below answers my question, but still curious about this.

Reply:
An add-on, by itself, has no channel - it only has versions that do. You can upload listed versions, or unlisted versions, or a mix of both, but can never change the channel for an existing version.

The flags we look at here are set on the add-on, but this is a check for a single version - and it's that version that is either listed or unlisted.